src/components/CitySearch.js

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+
+export const CitySearch = (props) => {
+  let handleInput = e => props.setCurrentCity(e.target.value);
+
+  return (
+      <form onSubmit={(e) => props.handleSubmit(e, props.currentCity)}>
+        <input type="text" value={props.currentCity} onChange={handleInput}/>
```

Comment:
It looks like you are reading from the current city in the handle submit function in the parent component. I'm wondering, what led you to update the current city on input change? Another approach that could be considered is keeping the city-state encapsulated in this component and only passing it back on submit. What are the tradeoffs there?

Reply:
I was previously using the city to display the city name in the parent component. Which is why I needed the city at the App level. I changed the design last minute. It could've have definitely been held in the CitySearch.js component. However, if in the future I wanted to use the city somewhere else I would've had to move it back up to the parent or implement another state management solution.